rpm/generic/zfs.spec.in

```diff
@@ -172,8 +172,12 @@ Obsoletes:      libzpool4
 This package contains the zpool library, which provides support
 for managing zpools
 
+%if 0%{?rhel} == 7 || 0%{?centos} == 7
```

Comment:
Pacemaker did this another way, with
```
%if %{defined ldconfig_scriptlets}
%ldconfig_scriptlets ...
%else
%post ...
%postun...
%endif
```
which might get some environments right that your method does not.  Just FYI.

Reply:
Updated.